src/include/buffer/buffer_pool_manager_instance.h

```diff
@@ -98,65 +114,32 @@ class BufferPoolManagerInstance : public BufferPoolManager {
    * Unset the dirty flag of the page after flushing.
    *
    * @param page_id id of page to be flushed, cannot be INVALID_PAGE_ID
-   * @return False if the page could not be found in the page table, true otherwise
+   * @return false if the page could not be found in the page table, true otherwise
    */
   auto FlushPgImp(page_id_t page_id) -> bool override;
 
   /**
    * TODO(P1): Add implementation
    *
-   * @brief Create a new page in the buffer pool.
-   *
-   * 1. Call AllocatePage() to get the page id for the new page
-   * 2. Pick a replacement page from either the free list or the replacer. (NOTE: always find from the free list first)
-   * 3. If the replacement page is dirty, write it back to disk by calling DiskManager::WritePage().
-   * 4. Delete the replacement page from the page table and insert the new page.
-   * 5. Update the new page metadata, zero out the memory and return the page pointer.
-   *
-   * Same as FetchPage(), remember to pin and record the access of the frame in the replacer.
-   *
-   * @param[out] page_id id of created page
-   * @return nullptr if no new pages could be created, otherwise pointer to new page
+   * @brief Flush all the pages in the buffer pool to disk.
    */
-  auto NewPgImp(page_id_t *page_id) -> Page * override;
+  void FlushAllPgsImp() override;
 
   /**
    * TODO(P1): Add implementation
    *
-   * @brief Delete a page from the buffer pool.
+   * @brief Delete a page from the buffer pool. If page_id is not in the buffer pool, do nothing and return true. If the
+   * pin count of page_id is greater than 0, don't delete and then return false.
```

Comment:
```suggestion
   * page is pinned and cannot be deleted return false immediately.
```

Reply:
I'm not sure on this one, was trying to follow how code comments are described in other assignments.